### DISCUSSION TITLE
More flexibility in linking to value from the comma separated string  in JSON file

### DISCUSSION THREAD

**Discussion by meiware** [Serial-Studio/Serial-Studio#41] [Ideas]
Thank you for the great application to get quick a good overview of all kind of data!

In my application (runs on STM32L4) I collect the data to visualize from multiple modules. Each module creates (a part of) the comma separated string by itself, because I don't want to have access to private data of the modules from a higher level in my application. Imagine I have three modules, if I want to add one more values to visualize to the comma separated string in the first module, I have to increment every following %n in my JSON file (I have over 60 values). So my idea is to support multiple comma separated strings which could be uniquely identified with a tag or something prefixed with a '@'. For each unique string the %n count start over. In JSON you can select a certain value with @[string tag]%[n'th value]

So example serialStudio input strings:
`/*@WP,[value 1],[value 2],[value 3],...*/`
`/*@BC,[value 1],[value 2],[value 3],...*/`
`/*@GG,[value 1],[value 2],[value 3],...*/`

Or if synchronization of the multiple comma seperated strings is a problem (but I think this is less preferred):
`/*@WP,[value 1],[value 2],[value 3],...,@BC,[value 1],[value 2],[value 3],...,@GG,[value 1],[value 2],[value 3],...*/`

Example JSON config file (fragment):
`              {
			"t":"General",
			"d":
                        [
				{
					"t":"Msg count",
					"v":"@WP%44"
				},
				{
					"t":"GG Data valid",
					"v":"@GG%1",
					"g":true
				},					
				{
					"t":"BC Data valid",
					"v":"@BC%1",
					"g":true
				},
				{
					"t":"WP Data valid",
					"v":"@WP%1",
					"g":true
				}		
			]
		},
		{
			"t":"Battery charger",
			"d":
                        [
				{
					"t":"BC Status",
					"v":"@BC%2"
				},
				{
					"t":"BC Temperature",
					"v":"@BC%3",
                                        "u":"C",
                                        "g":true
				}
                        ]
                },
		{
			"t":"Gas Gauge",
			"d":
                        [
				{
					"t":"GG Control stat",
					"v":"@GG%2"
				},
				{
					"t":"GG Flags",
					"v":"@GG%3"
				}
                        ]
                }
`

To stay compatible with the previous versions and thus no "@" is used, only one comma separated string is supported.

**Comment by alex-spataru** [comment]
As of the latest version, you can achieve this by customizing the [Frame Parser](https://github.com/Serial-Studio/Serial-Studio/wiki/Project-Editor#frame-parser-function-view) and writing your own JavaScript code to categorize the frames according to your specific requirements. It took me some time to get it right, but I believe this approach offers users the flexibility they need for their projects, while still retaining the simplicity and capability to handle most frame formats.
